Utils/Dataflow/017_adjustMetadata/adjustMetadata.py

```diff
@@ -163,32 +163,26 @@ def input_events(data):
     return result
 
 
-def transform_chain_data(data, original):
+def transform_chain_data(data):
     """ Transform chain_data into array of integers and get chain_id from it.
 
     chain_id is the taskid of the task chain's root.
 
     :param data: data to be updated, must contain taskid and proper chain_data
                  (string of numbers separated by commas).
     :type data: dict
-    :param original: string representing data at the beginning of the stage
-    :type original: str
 
     :return: True if update was successful, False otherwise (chain_id and
              chain_data will be updated so that the task is considered
              independent)
     :rtype: bool
     """
-    if type(data) is not dict:
-        sys.stderr.write('(WARN) Function transform_chain_data() received '
-                         'non-dict data: %s. Skipping.\n' % original)
-        return False
```

Comment:
There\`s nothing wrong in checking input parameter type, so I don\`t see why it was removed here.
"Pythonic" way, of course, is to handle exceptions, but I still prefer to check things...

When the check reveals invalid input data, it is OK to just say "we expected 'dict', got 'something_else'" (and/or, maybe, throw TypeError) -- so you don\`t actually need the original message here.

And, as you have mentioned somewhere: the function might be reused somewhere else, and no one guarantees that next time the outer code will check it in advance. And even here, to be honest: someone might decide to just remove this check from `process()` later, for some reason.